src/from_bytes_radix.rs

```diff
@@ -0,0 +1,425 @@
+/** A trait which provides `from_bytes_radix()` for integer types.
+
+This acts like `from_str_radix`, including panicking if `radix` is not in
+[2, 32] and supporting `[0-9A-Za-z]` as possible digits, depending on the
+value of `radix`.
+```
+use bstr::{FromBytesRadix, IntErrorKind, B};
+
+for radix in 2..=36 {
+
+    let r = radix as u8;
+
+    let e = B("");
+    let empty = u8::from_bytes_radix(&e, 10);
+    assert_eq!(empty.unwrap_err().kind(), &IntErrorKind::Empty);
+
+    let a = B("11");
+    let eleven = u8::from_bytes_radix(&a, radix);
+    assert_eq!(eleven, Ok(r + 1));
+
+    let b = B("111111111");
+    let pos_overflow = u8::from_bytes_radix(&b, radix);
+    assert_eq!(pos_overflow.unwrap_err().kind(), &IntErrorKind::PosOverflow);
+
+    let c = B("-111");
+    let negatory = u8::from_bytes_radix(&c, radix);
+    assert_eq!(negatory.unwrap_err().kind(), &IntErrorKind::InvalidDigit);
+
+
+    let r = radix as i32;
+
+    let totally_fine = i32::from_bytes_radix(&c, radix);
+    assert_eq!(totally_fine, Ok(-(r*r + r + 1)));
+}
+```
+
+The `NonZero` versions of integers are not currently supported for parsing.
+Instead, please parse the equivalent possibly-zero integer, then convert:
+```
+use core::num::NonZeroU8;
+use bstr::{FromBytesRadix, B};
+
+let a = B("11");
+let eleven = u8::from_bytes_radix(&a, 10).ok().and_then(NonZeroU8::new);
+assert_eq!(eleven, NonZeroU8::new(11));
+
+let zero = B("0");
+let nada = u8::from_bytes_radix(&zero, 10).ok().and_then(NonZeroU8::new);
+assert_eq!(nada, None);
+```
+
+*/
+
+pub trait FromBytesRadix {
```

Comment:
```suggestion
mod private {
    pub trait Sealed {}
}

pub trait FromBytesRadix: Sized + private::Sealed {
```

These changes remove a footgun of implementing this trait on an unsized type which would not give any methods and prevents this trait from being implemented by any other crates.